### ISSUE TITLE
Add PredictDigit REST Endpoint to Spring Boot API

### ISSUE THREAD

**Issue by TiffanyYongNgikChee** [TiffanyYongNgikChee/grpc-he-benchmark#38]
## Description

Add a `/api/predict` POST endpoint to the Spring Boot REST API that forwards digit prediction 
requests to the Rust gRPC server's `PredictDigit` RPC. This connects the frontend (React) to 
the encrypted inference engine running inside Docker.

The existing `GrpcClientService` already handles `runBenchmark()` and `runComparisonBenchmark()`. 
This issue adds a third method `predictDigit()` that calls the `PredictDigit` RPC defined in 
`he_service.proto`.

## Tasks

- [ ] Create `PredictRequest.java` model class with `pixels` (List<Long>, 784 values) and `scaleFactor` (long)
- [ ] Create `PredictResponse.java` model class with `predictedDigit`, `confidence`, `logits`, timing fields, and `status`
- [ ] Add `predictDigit()` method to `GrpcClientService.java` that calls `stub.predictDigit()`
- [ ] Add `@PostMapping("/api/predict")` endpoint to `BenchmarkController.java`
- [ ] Add error handling for when Docker container is not running
- [ ] Test with curl or Postman using a sample digit-7 pixel array

## Acceptance Criteria

- POST `/api/predict` with `{"pixels": [0,0,...255,...0], "scaleFactor": 1000}` returns JSON response
- Response includes `predictedDigit`, `confidence`, `logits` (10 values), and per-layer timing
- Returns proper error message (400/500) if pixels array is not exactly 784 values
- Works end-to-end: React → Spring Boot → gRPC → Docker → Encrypted CNN → Response

## Dependencies

- Requires Docker container running with `PredictDigit` RPC (already working ✅)
- Requires `he_service.proto` compiled for Java (already in `pom.xml` protobuf plugin)

## Files to Create/Modify

- `spring-boot-api/src/main/java/com/fyp/hebench/model/PredictRequest.java` (new)
- `spring-boot-api/src/main/java/com/fyp/hebench/model/PredictResponse.java` (new)
- `spring-boot-api/src/main/java/com/fyp/hebench/service/GrpcClientService.java` (add predictDigit method)
- `spring-boot-api/src/main/java/com/fyp/hebench/controller/BenchmarkController.java` (add /api/predict endpoint)

## How to Test

```bash
# 1. Start Docker container (Rust gRPC server)
docker run -p 50051:50051 he-benchmark-server

# 2. Start Spring Boot
cd spring-boot-api && mvn spring-boot:run

# 3. Test with curl
curl -X POST http://localhost:8080/api/predict \
  -H "Content-Type: application/json" \
  -d '{"pixels": [0,0,0,...784 values...], "scaleFactor": 1000}'
